R/Model-class.R

```diff
@@ -931,338 +931,224 @@ LogisticLogNormalMixture <- function(mean,
+#'   the Beta prior on the transformation `kappa = (rho + 1) / 2`, which is in
+#'   (0, 1). For example, `a = 1, b = 1` leads to a uniform prior on `rho`.
+#' @slot use_fixed rho (`logical`)\cr indicates whether a fixed value for
+#'   `sigma2W` and `rho` (for each parameter separately) is used or not. This
+#'   slot is needed for internal purposes and must not be touched by the user.
+#'
+#' @seealso [`DualEndpointRW`], [`DualEndpointBeta`], [`DualEndpointEmax`].
+#'
+#' @aliases DualEndpoint
+#' @export
+#'
+.DualEndpoint <- setClass(
+  Class = "DualEndpoint",
+  slots = c(
+    betaZ_params = "ModelParamsNormal",
+    ref_dose = "numeric",
+    use_log_dose = "logical",
+    sigma2W = "numeric",
+    rho = "numeric",
+    use_fixed = "logical"
+  ),
+  prototype = prototype(
+    probit_params = ModelParamsNormal(
+      mean = c(0, 1),
+      cov = diag(2)
+    ),
+    ref_dose = 1,
+    use_log_dose = FALSE,
+    sigma2W = 1,
+    rho = 0,
+    use_fixed = c(
+      sigma2W = TRUE,
+      rho = TRUE
+    )
+  ),
+  contains = "Model",
+  validity = v_model_dual_endpoint
+)
 
-                     o$result()
-                 })
+## constructor ----
 
-##' Initialization function for the "DualEndpoint" class
-##'
-##' @param mu see \code{\linkS4class{DualEndpoint}}
-##' @param Sigma see \code{\linkS4class{DualEndpoint}}
-##' @param refDose see \code{\linkS4class{DualEndpoint}} (default: 1)
-##' @param useLogDose see \code{\linkS4class{DualEndpoint}}
-##' (default: \code{FALSE})
-##' @param sigma2W see \code{\linkS4class{DualEndpoint}}
-##' @param rho see \code{\linkS4class{DualEndpoint}}
-##' @return the \code{\linkS4class{DualEndpoint}} object
-##'
-##' @export
-##' @keywords methods
-DualEndpoint <- function(mu,
-                         Sigma,
-                         refDose=1,
-                         useLogDose=FALSE,
+#' @rdname DualEndpoint-class
+#'
+#' @param mean (`numeric`)\cr for the probit toxicity model, the prior mean vector.
+#' @param cov (`matrix`)\cr for the probit toxicity model, the prior covariance
+#'   matrix. The precision matrix is internally calculated as an inverse of `cov`.
+#' @param ref_dose (`number`)\cr for the probit toxicity model, the reference dose.
+#' @param use_log_dose (`flag`)\cr for the probit toxicity model, whether a log
+#'   transformation of the (standardized) dose should be used?
+#' @param sigma2W (`numeric`)\cr either a fixed value for the biomarker variance,
+#'   or a numeric vector with elements named `a` and `b` for the Inverse-Gamma
+#'   prior parameters.
+#' @param rho (`numeric`)\cr either a fixed value for the correlation
+#'   (between -1 and 1), or a named vector with elements named `a` and `b` for
+#'   the Beta prior on the transformation `kappa = (rho + 1) / 2`, which is in
+#'   (0, 1). For example, `a = 1, b = 1` leads to a uniform prior on `rho`.
+#'
+#' @export
+#'
+DualEndpoint <- function(mean,
+                         cov,
```

Comment:
I wonder here since I am not too worried about backwards compatibility whether we should directly take the `ModelParamsNormal` instead of separate `mean` and `cov`?

Reply:
Sure, we could do this, but that would not only be a backward compatibility issue (which is not the case as you say) but also inconsistency with other (normal) Model constructors.....

Reply:
ok, then let's keep as is